src/genai/prompt/prompt_service.py

```diff
@@ -190,9 +192,9 @@ def list(
             limit=limit,
             offset=offset,
             search=search,
-            task_id=task_id,
-            model_id=model_id,
-            source=[to_enum(PromptRetrieveRequestParamsSource, s) for s in cast_list(source)] if source else None,
+            task_id=to_list_if_sequence(task_id),
+            model_id=to_list_if_sequence(model_id),
```

Comment:
`to_list_if_sequence` function is needed to not convert str to list[str] but only sequence[str] to list